server/render/index.js

```diff
@@ -74,8 +75,40 @@ export function render( element, key = JSON.stringify( element ) ) {
 	//todo: render an error?
 }
 
+function getCacheKey( context ) {
+	if ( ! isSectionIsomorphic( context.store.getState() ) || context.user ) {
+		return false;
+	}
+
+	if ( isEmpty( context.query ) ) {
+		return context.pathname;
+	}
+
+	let queryParams = Object.keys( context.query );
+	const cacheConfiguration = get( getIsomorphicSectionConfiguration( context.store.getState() ), 'cache', null );
+
+	if ( cacheConfiguration && cacheConfiguration.queryParams === true ) {
+		return context.canonicalPath;
```

Comment:
Oh, actually, since this is for the case where no `cacheConfiguration` is specified (e.g. themes), we should fall back to the query args-free `pathname` anyway, shouldn't we?